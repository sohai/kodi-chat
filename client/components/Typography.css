.root {
  font-family: "Roboto", sans-serif;
}
.light {
  color: #fff;
}
.dark {
  color: rgba(255, 255, 255, 0.87);
}
.display1 {
  font-size: 2rem;
}
.body {
  font-size: 1rem;
}
