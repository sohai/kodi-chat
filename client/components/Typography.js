import React from 'react';
import PropTypes from 'prop-types';
import styles from './Typography.css';
import classNames from 'classnames';

const propTypes = PropTypes && {
  color: PropTypes.oneOf(['light', 'dark']),
  variant: PropTypes.oneOf(['display1', 'body']),
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  children: PropTypes.node
};

const defaultProps = {
  color: 'dark',
  variant: 'body',
  component: 'div'
};

const Typography = ({
  className: classNameProp,
  color,
  variant,
  component: Component,
  ...rest
}) => {
  const className = classNames(
    styles.root,
    {
      [styles.light]: color === 'light',
      [styles.dark]: color === 'dark',
      [styles.display1]: variant === 'display1'
    },
    classNameProp
  );

  return <Component className={className} {...rest} />;
};
Typography.displayName = 'Typography';
Typography.propTypes = propTypes;
Typography.defaultProps = defaultProps;
export default Typography;
