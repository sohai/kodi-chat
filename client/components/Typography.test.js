/* global shallow, describe, it, expect */
/* eslint-disable import/first */
import React from 'react';
import Typography from './Typography';

describe('(Component) TextInput', () => {
  it('should render with default props', () => {
    const props = {
      children: 'Typography default'
    };
    const wrapper = shallow(<Typography {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with given color variant and component', () => {
    const props = {
      children: 'Typography with props',
      color: 'light',
      variant: 'display1',
      component: 'span'
    };
    const wrapper = shallow(<Typography {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
