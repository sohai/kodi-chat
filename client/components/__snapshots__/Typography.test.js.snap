// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TextInput should render with default props 1`] = `
<div
  className="root dark"
>
  Typography default
</div>
`;

exports[`(Component) TextInput should render with given color variant and component 1`] = `
<span
  className="root light display1"
>
  Typography with props
</span>
`;
